lib/src/analysis/util.rs

```diff
@@ -0,0 +1,7 @@
+macro_rules! unpack {
+    ($pat:pat = $val:expr) => {
+        let $pat = $val else { return None; };
+    };
+}
+
+pub(crate) use unpack;
```

Comment:
What the hell are you talking about Jessie? Please explain to me what this does?

Reply:
lmao i had to google this one, it's apparently the only way to make a macro usable from a module

Reply:
added comments explaining it